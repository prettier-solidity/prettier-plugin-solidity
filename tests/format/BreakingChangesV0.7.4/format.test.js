runFormatTest(import.meta, ['slang-solidity'], { compiler: '0.7.4' });
runFormatTest(import.meta, ['slang-solidity'], {
  compiler: '0.7.4',
  bracketSpacing: true
});
runFormatTest(import.meta, ['slang-solidity'], { compiler: '0.7.3' });
runFormatTest(import.meta, ['slang-solidity'], {
  compiler: '0.7.3',
  bracketSpacing: true
});
