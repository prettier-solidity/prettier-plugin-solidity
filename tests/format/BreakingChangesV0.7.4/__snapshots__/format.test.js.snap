// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreakingChangesV0.7.4.sol - {"bracketSpacing":true} format 1`] = `
====================================options=====================================
bracketSpacing: true
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import {symbol1 as alias1, symbol2 as alias2, symbol3 as alias3, symbol4} from "File2.sol";

=====================================output=====================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import { symbol1 as alias1, symbol2 as alias2, symbol3 as alias3, symbol4 } from "File2.sol";

================================================================================
`;

exports[`BreakingChangesV0.7.4.sol - {"compiler":"0.7.3","bracketSpacing":true} format 1`] = `
====================================options=====================================
bracketSpacing: true
compiler: "0.7.3"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import {symbol1 as alias1, symbol2 as alias2, symbol3 as alias3, symbol4} from "File2.sol";

=====================================output=====================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import { symbol1 as alias1, symbol2 as alias2, symbol3 as alias3, symbol4 } from "File2.sol";

================================================================================
`;

exports[`BreakingChangesV0.7.4.sol - {"compiler":"0.7.3"} format 1`] = `
====================================options=====================================
compiler: "0.7.3"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import {symbol1 as alias1, symbol2 as alias2, symbol3 as alias3, symbol4} from "File2.sol";

=====================================output=====================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import {symbol1 as alias1, symbol2 as alias2, symbol3 as alias3, symbol4} from "File2.sol";

================================================================================
`;

exports[`BreakingChangesV0.7.4.sol - {"compiler":"0.7.4","bracketSpacing":true} format 1`] = `
====================================options=====================================
bracketSpacing: true
compiler: "0.7.4"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import {symbol1 as alias1, symbol2 as alias2, symbol3 as alias3, symbol4} from "File2.sol";

=====================================output=====================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import {
    symbol1 as alias1,
    symbol2 as alias2,
    symbol3 as alias3,
    symbol4
} from "File2.sol";

================================================================================
`;

exports[`BreakingChangesV0.7.4.sol - {"compiler":"0.7.4"} format 1`] = `
====================================options=====================================
compiler: "0.7.4"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import {symbol1 as alias1, symbol2 as alias2, symbol3 as alias3, symbol4} from "File2.sol";

=====================================output=====================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import {
    symbol1 as alias1,
    symbol2 as alias2,
    symbol3 as alias3,
    symbol4
} from "File2.sol";

================================================================================
`;

exports[`BreakingChangesV0.7.4.sol format 1`] = `
====================================options=====================================
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import {symbol1 as alias1, symbol2 as alias2, symbol3 as alias3, symbol4} from "File2.sol";

=====================================output=====================================
pragma solidity ^0.7.0;

/**
 * - if options.compiler is undefined we won't break the ImportDirective.
 * - if options.compiler is lower than 0.7.4 we won't break the ImportDirective.
 * - if options.compiler is greater than or equal to 0.7.4 we will break the
 *   ImportDirective if the line length exceeds the printWidth.
 */
import {symbol1 as alias1, symbol2 as alias2, symbol3 as alias3, symbol4} from "File2.sol";

================================================================================
`;
